app/validation/validator.py

```diff
@@ -277,6 +246,111 @@ def _get_answers_for_block(block_json):
                 answers.append(answer_json)
         return answers
 
+    @staticmethod
+    def _get_answer_minimum(answer, answer_ranges):
+        defined_minimum = answer.get('min_value')
+        minimum = None
+
+        if defined_minimum is None:
+            minimum = [0]
+        elif 'value' in defined_minimum:
+            minimum = [defined_minimum.get('value')]
+        elif 'answer_id' in defined_minimum:
+            referred_answer = answer_ranges.get(defined_minimum['answer_id'])
+            if referred_answer is None:
+                minimum = None  # Referred answer is not  valid (picked up by _validate_referred_numeric_answer)
+            else:
+                minimum = referred_answer['min'] + referred_answer['max']
+
+        if defined_minimum and defined_minimum.get('exclusive') and minimum:
+            exclusive_minimum = []
+            for value in minimum:
+                exclusive_minimum.append(value + (1 / 10 ** answer.get('decimal_places', 0)))
+            minimum = exclusive_minimum
+
+        return minimum
+
+    @staticmethod
+    def _get_answer_maximum(answer, answer_ranges):
```

Comment:
there is a lot of common code between this and `_get_answer_minimum` any chance of calling a 3rd function to do the common logic?

Reply:
@jonnyshaw89 I've split it out so see what you think. To simplify it i've done it in two steps. One that gets the defined value, then another to convert that to exclusive if required.